soc/arm/renesas_rcar/common/clock_rcar.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 IoT.bzh
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#ifndef ZEPHYR_SOC_ARM_RENESAS_RCAR_COMMON_CLOCK_RCAR_H_
+#define ZEPHYR_SOC_ARM_RENESAS_RCAR_COMMON_CLOCK_RCAR_H_
+
+#include <dt-bindings/clock/renesas/renesas-cpg-mssr.h>
+
+/**
+ * @brief Enable or disable a core clock
+ *
+ * On success, the clock is enabled or disabled.
+ *
+ * @param base_address Clock controller base address
+ * @param module Clock module number
+ * @param rate Target rate for clock to be endisable
+ * @param enable 1 to enable a clock, 0 to disable it
+ *
+ * @retval 0 on success
+ * @retval -EINVAL if clock or rate not supported
+ */
+int soc_cpg_core_clock_endisable(uint32_t base_address, uint32_t module,
```

Comment:
why is this code moved under `soc/`? why can't it be under `drivers/clock_control` (I guess driver can have its own folder with multiple sources if it helps).

Reply:
Hi @gmarull, thanks for your review ! 
I moved these part of code into `/soc` for multiples reasons:

- By creating a driver file for each SoC, this `/driver/clock_control/renesas` folder (and so Zephyr sources) could drastically grow in a way that could be hard to maintain.
Each new SoC added (3 SoCs are already on the way) would add their own clock driver that would duplicate a lot of common code with other SoC ones.

- Indeed, especially for renesas SoCs, clocks trees are SoC specific, so, when we will add new SoC in the future, it will be easier to add the associated implementation in the newly created `/soc` folder instead of doing changes to drivers. Also, the git history would be much cleaner this way (by not editing both `/soc` and `/driver` domain).

- This rework and the choice of the `/soc` directory also follows the work concerning the pinctrl driver for Renesas rcar SoCs (#40550), I copied the way that this implementation is creating a source code per SoC in the `/soc` directory to implement SoC specific part of the code for the sake of homogeneity. 

- The hierarchy created with clock_rcar.h (renesas) -> clock_soc.h (gen3) -> cpg_r8a..c (soc) files is also allowing us to duplicate the least possible code in the Zephyr code base. This scheme would not be as clean as it is if moved under the `/driver` folder.

Reply:
We (Aymeric :) ) will give a try with a reversed logic.
One driver per SoC under a drivers/clock_control/renesas directory, which will be enabled based on SoC config value.
that way the current clock_control_rcar_cpg_mssr.c will export some common function that will be used by the different driver.